csharp/src/Drivers/BigQuery/BigQueryStatement.cs

```diff
@@ -51,7 +51,26 @@ public BigQueryStatement(BigQueryClient client, GoogleCredential credential)
 
         public override QueryResult ExecuteQuery()
         {
-            QueryOptions? queryOptions = ValidateOptions();
+            QueryOptions queryOptions = ValidateOptions();
+
+            if (this.client.ProjectId == BigQueryConstants.DetectProjectId)
```

Comment:
Wouldn't the same logic also be needed for `ExecuteUpdate`?

Reply:
added in latest push